base.py

```diff
@@ -238,44 +241,47 @@ def predict_sequentially_all(self, X, label_col_names: List[str] = None):
                 f_name for f_name in feats if f_name in label_col_names
             ]
             # initialize feat_dict to first row & pred_dict to match first row too
-            feat_dict = OrderedDict(list(zip(feats, X[0])))
+            feat_dict = OrderedDict(zip(feats, list(X[0])))
             pred_dict = dict(
                 [(k, v) for (k, v) in feat_dict.items() if k in preds_that_are_feats]
             )
 
             # sequentially iterate retriving next prediction based on previous prediction
             preds = []
             for i in range(len(X)):
-                # extrac next row feats
-                feat_dict = OrderedDict(list(zip(feats, list(X[i]))))
-                # update feats with previous prediction
-                for f_name in preds_that_are_feats:
-                    feat_dict[f_name] = pred_dict[f_name]
+                for (f_name, x_value) in zip(feats, list(X[i])):
+                    if f_name in preds_that_are_feats:
+                        if not self.diff_state:
+                            feat_dict[f_name] = pred_dict[f_name]
+                        else:
+                            feat_dict[f_name] += pred_dict[f_name]
```

Comment:
For the rest of loop it is okay. But, the 1st row is same for feat_dict and pred_dict, so adding to 1st `feat_dict[0]` will be `feat_dict[0]+feat_dict[0]`, as `pred_dict[0]=feat_dict[0]` in line 246. Should this step be skipped for i=0? In that case, you are considering st+1-st =0 for 1st step. Alternative is to have a very small value for it (since we don't have a past in the 1st step/row at i=0). If we have already thrown away X[0], then using the feat_dict[1]-feat_dict[0] may also be okay.

Reply:
Fair catch! I have changed initialization of "pred_dict" to be all zeroes on last commit, sot as to keep the innerloop functionality the same.

Reply:
Thanks. That works.